manifests/init.pp

```diff
@@ -72,10 +73,20 @@
       refreshonly => true,
     }
 
-    create_resources('firewalld_port',      $ports)
-    create_resources('firewalld_zone',      $zones)
-    create_resources('firewalld_service',   $services)
-    create_resources('firewalld_rich_rule', $rich_rules)
+    $defaults = { zone => $default_zone }
+
+    if $zones {
+      create_resources('firewalld_zone', $zones)
+    }
+    if $ports {
```

Comment:
The if statements around all of these are not really needed - if create_resources recieves an empty hash it just won't do anything - adding resource defaults doesn't alter this behaviour.